AraCorrelator/RayTraceCorrelator.cxx

```diff
@@ -162,78 +133,56 @@ void RayTraceCorrelator::LoadArrivalTimeTables(const std::string &filename, int
         sprintf(errorMessage, "Opening of the table (%s) was unsuccessful\n",filename.c_str());
         throw std::runtime_error(errorMessage);
     }
-    
-    // then, try to load the arrival times tables
-    TTree * tTree = (TTree*) infile->Get("tArrivalTimes");
-    if(!tTree){
-        infile->Close();
-        sprintf(errorMessage, "Arrival time tables could not be found in the file (%s)\n",filename.c_str());
-        throw std::runtime_error(errorMessage);
-    }
 
-    // if the file is good, load the variables from the branches
-    int ant, phiBin, thetaBin;
-    double phi, theta;
-    double arrivalTime, arrivalTheta, arrivalPhi, launchTheta, launchPhi;
-    tTree -> SetBranchAddress("ant", & ant);
-    tTree -> SetBranchAddress("phiBin", & phiBin);
-    tTree -> SetBranchAddress("thetaBin", & thetaBin);
-    tTree -> SetBranchAddress("arrivalTime", & arrivalTime);
-    tTree -> SetBranchAddress("arrivalTheta", & arrivalTheta);
-    tTree -> SetBranchAddress("arrivalPhi", & arrivalPhi);
-    tTree -> SetBranchAddress("launchTheta", & launchTheta);
-    tTree -> SetBranchAddress("launchPhi", & launchPhi);    
-    tTree -> SetBranchAddress("phi", & phi);
-    tTree -> SetBranchAddress("theta", & theta);
-
-    // and put those values into the vector
-    int nEntries = tTree -> GetEntries();
-    for (int i = 0; i < nEntries; i++) {
-        tTree -> GetEntry(i);
-        arrivalTimes_[solNum][thetaBin][phiBin][ant] = arrivalTime;
-        arrivalThetas_[solNum][thetaBin][phiBin][ant] = arrivalTheta;
-        arrivalPhis_[solNum][thetaBin][phiBin][ant] = arrivalPhi;
-        launchThetas_[solNum][thetaBin][phiBin][ant] = launchTheta;
-        launchPhis_[solNum][thetaBin][phiBin][ant] = launchPhi;        
+    std::map<int, TH2D> this_arrvialTimes;
+    std::map<int, TH2D> this_arrivalThetas;
+    std::map<int, TH2D> this_arrivalPhis;
+    std::map<int, TH2D> this_launchThetas;
+    std::map<int, TH2D> this_launchPhis;
+
+    for(int i=0; i<numAntennas_; i++){
+        std::   stringstream ss;
+
+        ss.str(""); ss << "arrival_time_ch_" << i;
+        TH2D *hArrivalTime = (TH2D*)((TH2D*)infile->Get(ss.str().c_str())->Clone()); // clone, so this survives the file closure
+        this_arrvialTimes[i] = *hArrivalTime; // de ref the pointer; give me the memory
```

Comment:
What this is allowed?

Reply:
Sure, why not? :)

Reply:
(Got the idea from here: https://root-forum.cern.ch/t/filling-a-th2d-in-map-pair-int-int-th2d/24791)

Reply:
I'm just confused because the TH2D assignment operator is private. But if it's compiling it's compiling.